src/lava/lib/optimization/utils/solver_tuner.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+import itertools as it
+import typing as ty
+
+from lava.lib.optimization.solvers.generic.solver import OptimizationSolver
+import random
+
+
+class SolverTuner:
+    """Class to find and set hyperparameters for an OptimizationSolver."""
+
+    def __init__(self, params_grid: ty.Dict):
+        """
+        Instantiate a SolverTuner, defining the admissible search space.
+
+        Parameters
+        ----------
+        params_grid: dict
+            Dictionary where the keys are the name of the hyperparameters to be
+            tuned, and the values are tuples containing all the admissible
+            values for the associated hyperparameter.
+        """
+        self._params_grid = params_grid
+
+    def tune(self,
+             solver: OptimizationSolver,
+             solver_parameters: ty.Dict,
+             stopping_condition: ty.Callable[[float, int], bool] = None):
+        """Find and set solving hyperparameters for an OptimizationSolver"""
+        # TODO : Check that hyperparameters are arguments for solver
+        best_hyperparameters, best_cost, best_step_to_sol = self._random_search(
+            solver,
+            solver_parameters,
+            stopping_condition)
+        succeeded = stopping_condition(best_cost, best_step_to_sol)
+        return best_hyperparameters, succeeded
+
+    def _random_search(self,
+                       solver: OptimizationSolver,
+                       solver_parameters: ty.Dict,
+                       stopping_condition: ty.Callable[
+                           [float, int], bool] = None
+                       ) -> ty.Union[ty.NoReturn, ty.Dict]:
+
+        best_hyperparameters = None
+        best_cost = float("inf")
+        best_step_to_sol = float("inf")
+        problem = solver.problem
+        params_names = self._params_grid.keys()
+        params_grid = list(
+            it.product(*map(lambda k: self._params_grid[k], params_names)))
+        random.shuffle(params_grid)

```

Comment:
Interesting, is there a reason why you moved from a grid search to a random shuffling? Asking out of curiosity.

Reply:
We expect this to be faster, random sampling would be better than brute force, in particular if you start your grid search in a bad point of the configuration space. Bad in the sense of a dynamics that does not converge to a solution (or at all).